src/main/kotlin/org/rust/ide/annotator/cargoCheck/RsCargoCheckUtils.kt

```diff
@@ -320,15 +317,21 @@ private fun formatMessage(message: String): String {
  * these documents should be stripped later (when [saveAllDocuments] is called).
  */
 private fun saveAllDocumentsAsTheyAre() {
+    checkWriteAccessAllowed()
     val documentManager = FileDocumentManager.getInstance()
-    for (document in documentManager.unsavedDocuments) {
+    val rustfmtWatcher = ApplicationManager.getApplication().getComponent(RustfmtWatcher::class.java)
+    val rustfmtWatcherStatus = rustfmtWatcher.isEnabled
+    rustfmtWatcher.isEnabled = false
```

Comment:
Maybe it's better to create `RustfmtWatcher#withoutReformating` method which takes lambda and move all logic related to changes in `RustfmtWatcher` state into `RustfmtWatcher` itself
It should provide a simpler and prettier api